pkg/reconciler/v1alpha1/route/resources/service.go

```diff
@@ -17,14 +17,77 @@ limitations under the License.
 package resources
 
 import (
-	"github.com/knative/pkg/kmeta"
-	"github.com/knative/serving/pkg/apis/serving/v1alpha1"
-	"github.com/knative/serving/pkg/reconciler/v1alpha1/route/resources/names"
+	"errors"
+	"fmt"
 
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/knative/pkg/kmeta"
+	netv1alpha1 "github.com/knative/serving/pkg/apis/networking/v1alpha1"
+	"github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	"github.com/knative/serving/pkg/reconciler/v1alpha1/route/resources/names"
 )
 
+var errLoadBalancerNotFound = errors.New("failed to fetch loadbalancer domain/IP from ingress status")
+
+// NewMakeK8sService creates a Service that redirect to the loadbalancer specified
+// in ClusterIngress status. It's owned by the provided v1alpha1.Route.
+// The purpose of this service is to provide a domain name for Istio routing.
+// TODO: It is to replace the "MakeK8sService" func after route reconciler is switched to reconcile
+// ClusterIngress.
+func NewMakeK8sService(route *v1alpha1.Route, ingress *netv1alpha1.ClusterIngress) (*corev1.Service, error) {
```

Comment:
Here I rename the new func to avoid conflict.
Will remove the old one once we finally switch to the new pattern in reconciler.